src/Language/TSX/Resolution.hs

```diff
@@ -0,0 +1,128 @@
+  where
+    notFound xs = throwResolutionError $ NotFoundError relImportPath xs Language.TypeScript
+
+-- | Resolve a non-relative TypeScript import to a known 'ModuleName' or fail.
+--
+-- import { b } from "moduleB" in source file /root/src/moduleA.ts
+--
+-- /root/src/node_modules/moduleB.ts
+-- /root/src/node_modules/moduleB/package.json (if it specifies a "types" property)
+-- /root/src/node_modules/moduleB/index.ts
+--
+-- /root/node_modules/moduleB.ts, etc
+-- /node_modules/moduleB.ts, etc
+resolveNonRelativePath :: ( Member (Modules address value) sig
+                          , Member (Reader M.ModuleInfo) sig
+                          , Member (Reader PackageInfo) sig
+                          , Member (Reader Span) sig
+                          , Member (Resumable (BaseError ResolutionError)) sig
+                          , Member Trace sig
+                          , Carrier sig m
+                          )
+                       => FilePath
+                       -> [String]
+                       -> Evaluator term address value m M.ModulePath
+resolveNonRelativePath name exts = do
+  M.ModuleInfo{..} <- currentModule
+  go "." modulePath mempty
+  where
+    nodeModulesPath dir = takeDirectory dir </> "node_modules" </> name
+    -- Recursively search in a 'node_modules' directory, stepping up a directory each time.
+    go root path searched = do
+      trace ("attempting to resolve (non-relative) require/import " <> show name)
+      res <- resolveModule (nodeModulesPath path) exts
+      case res of
+        Left xs | parentDir <- takeDirectory path , root /= parentDir -> go root parentDir (searched <> xs)
+                | otherwise -> notFound (searched <> xs)
+        Right m -> m <$ traceResolve name m
+    notFound xs = throwResolutionError $ NotFoundError name xs Language.TypeScript
+
+-- | Resolve a module name to a ModulePath.
+resolveModule :: ( Member (Modules address value) sig
+                 , Member (Reader PackageInfo) sig
+                 , Member Trace sig
+                 , Carrier sig m
+                 )
+              => FilePath -- ^ Module path used as directory to search in
+              -> [String] -- ^ File extensions to look for
+              -> Evaluator term address value m (Either [FilePath] M.ModulePath)
+resolveModule path' exts = do
+  let path = makeRelative "." path'
+  PackageInfo{..} <- currentPackage
+  let packageDotJSON = Map.lookup (path </> "package.json") packageResolutions
+  let searchPaths =  ((path <.>) <$> exts)
+                  <> maybe mempty (:[]) packageDotJSON
+                  <> (((path </> "index") <.>) <$> exts)
+  trace ("searching in " <> show searchPaths)
+  maybe (Left searchPaths) Right <$> resolve searchPaths
+
+typescriptExtensions :: [String]
+typescriptExtensions = ["ts", "tsx", "d.ts"]
+
+javascriptExtensions :: [String]
+javascriptExtensions = ["js"]
```

Comment:
Are there differences between this `Resolution` module and `TypeScript.Resolution`? I'm curious if we can still maintain a single `Resolution` module for `TypeScript` and `TSX`?